documentation/mqtt-overview.md

```diff
@@ -0,0 +1,88 @@
+
+### Update Behavior
+
+As long as a M5 is powered on, it publishes updates to the broker servers under the `/phone/maker/YOUR_SERIAL_NUMBER_HERE/notice` topic as a client device. It only publishes an update when a value changes.
+
+The M5 also subscribes to other topics, where it will react to events, and treat them as commands to perform various actions. In this way, the M5 can be partially remote-controlled over MQTT.
+
+Another client device (your phone or computer) subscribes to topics associated with the target printer and can publish commands for execution on the target printer.
+
+The messages are exchanged via an encrypted tunnel (TLS) and that's standard per the MQTT specification. What's unusual about AnkerMake MQTT messages, is that part of the payload is also encrypted before transmission using standard AES-256-CBC. In order for the communications to be successful, the messages must be encrypted and decrypted at either end. The AES encryption of the payload is redundant from a security standpoint (since the entire connection is already encrypted), but it's required for any program interfacing with an M5 to function.
+
+In order to encrypt and decrypt the payload, it's necessary to utilize the AnkerMake HTTPS API. The API allows a client to pull down the MQTT login info and MQTT-only encryption key needed to setup a new client device connection instance and perform cryptography on the payload. 
+
+Your **Authentication Token** (auth_token) is needed to authenticate a client's identity with the AnkerMake HTTPS servers. Then those MQTT credentials are presented to the MQTT server to subscribe/publish to the topics in the broker. Finally the MQTT-only key is used to decrypt incoming messages and encrypt outgoing messages.
+
+The basic relationships for AnkerMake MQTT communications can be illustrated as such:
+
+```mermaid
+graph
+    A("[Broker] Ankermake MQTT Servers (Job Queue and Data)") <-- "MQTT Data (TLS + Encrypted Palyoad)" --> B(("M5 Printers"))
+    A <-- "MQTT Data (TLS + Encrypted Palyoad)" --> C(("Client Devices"))
+    D("AnkerMake HTTPS Servers (API)") -- "MQTT Login Info + Key (HTTPS)" --> C
+    C -- "Authentication Token (HTTPS)" --> D
+```
+
+### Payload Structure
+
+As detailed in the previous section, part of the payload is encrypted. The payload is split into two parts: an unencrypted header and an encrypted JSON body. A MQTT-only key is used to decrypt incoming messages and encrypt outgoing messages per AES-256-CBC. This parsing and cryptography can be accomplished in your programs by using the `libflagship` library in this repository.
+
+In the `specifications` folder, you'll find a file named `mqtt.stf` that maps out the contents of the message payload. There's a few "[magic bytes](https://www.netspi.com/blog/technical/web-application-penetration-testing/magic-bytes-identifying-common-file-formats-at-a-glance/)" in there, but most of the application-specific JSON body is mapped out.
```

Comment:
Ah, we have some confusing terminology here.

In *many* cases (in libflagship), the purpose of `magic` bytes are entirely known.

The meaning of "Magic" bytes come from convention/decree, rather than value.

For example, PPPP frames starts with `0xF1`. Why `0xF1`? No particular reason, but that's what they chose. So it's "magic" (but known).

Contrast this with other bytes that might encode a length, size, time, loudness, color or similar value type. Those bytes have a fairly predictable meaning, and have very little arbitrary variation.

Reply:
I understand that but my explanation was confusing. I'm just going to remove that part as it's not really necessary to explain here.